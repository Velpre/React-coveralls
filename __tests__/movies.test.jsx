import React from "react";
import ReactDOM from "react-dom";

import {ListMovies, NewMovieForm} from "../movieApplication";
import {Simulate} from "react-dom/test-utils";

describe("movies application", () => {

    it("shows movie list", () => {
        const element = document.createElement("div");
        ReactDOM.render(<ListMovies />, element);
        expect(element.querySelector("h1").innerHTML)
            .toEqual("List movies");
        expect(element.innerHTML).toMatchSnapshot();
    });
    it("show new movie form", () => {
        const element = document.createElement("div");
        ReactDOM.render(<NewMovieForm />, element);
        expect(element.innerHTML).toMatchSnapshot();
    })

    it("submit new movie", () => {
        const onAddMovie = jest.fn(); // "Mock function" ("tullefunksjon")
        // Denne tar opp kallet i onSubmit og tar var på det slik at vi senere kan spørre hvordan onAddMovie ble brukt

        // Senere i testen skal vi skrive expect(onAddMovie).toHaveBeenCalled()


        const element = document.createElement("div");
        ReactDOM.render(<NewMovieForm onAddMovie={onAddMovie} />, element);

        Simulate.change(
            // finn elementet med date-testid=title og send en input event til det
            element.querySelector("[data-testid=title]"),
            {target: {value: "Movie 1"}}
        );
        Simulate.change(
            // finn elementet med date-testid=year og send en input event til det
            element.querySelector("[data-testid=year]"),
            {target: {value: "2022"}}
        );
        // finn elementet av type form og send en submit event til det
        Simulate.submit(element.querySelector("form"));

        expect(onAddMovie).toHaveBeenCalledWith({
            title: "Movie 1", year: "2022", plot: ""
        })
    })

})