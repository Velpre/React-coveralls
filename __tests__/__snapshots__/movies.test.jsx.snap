// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`movie pages shows movie list 1`] = `"<div><h1>Movies</h1><div class=\\"movie\\"><h2>Don't look up (2021)</h2><p>Impending disaster, but will politicians act?</p></div><div class=\\"movie\\"><h2>Web development (2022)</h2><p>Johannes Codes, everyone is confused</p></div></div>"`;

exports[`movie pages shows movies list 1`] = `"<div><h1>Movies</h1><div class=\\"movie\\"><h2>Don't look up (2021)</h2><p>Impending disaster, but will politicians act?</p></div><div class=\\"movie\\"><h2>Web development (2022)</h2><p>Johannes Codes, everyone is confused</p></div></div>"`;

exports[`movies application show new movie form 1`] = `"<form><h1>Add new movie</h1><div>Title: <input data-testid=\\"title\\" value=\\"\\"></div><div>Year: <input data-testid=\\"year\\" value=\\"\\"></div><div>Plot:<div><textarea data-testid=\\"plot\\"></textarea></div></div></form>"`;

exports[`movies application shows movie list 1`] = `"<div><h1>List movies</h1><ul><li>Movie 1</li><li>Movie 2</li></ul></div>"`;
